lua/entities/gmod_wire_expression2/core/custom/ftrace.lua

```diff
@@ -253,72 +288,196 @@ local function trcWorld(oFTrc, eE, vP, vA)
 	local trS, trE = oFTrc.mTrI.start, oFTrc.mTrI.endpos
 	trS:Set(eP); trE:Set(eA:Forward())
 	trE:Mul(oFTrc.mLen); trE:Add(trS)
-	-- http://wiki.garrysmod.com/page/util/TraceLine
+	-- https://wiki.facepunch.com/gmod/util.TraceLine
 	util.TraceLine(oFTrc.mTrI); return oFTrc
 end
 
-local function dumpItem(oFTrc, sNam, sPos)
+local function updateEarSize(oFTrc)
+	if(not oFTrc) then return nil end
+	local tE, iE = oFTrc.mFlt.Ear, 1
+	while(tE[iE]) do local vE = tE[iE]
+		if(isValid(vE)) then iE = iE + 1
+		else table.remove(tE, iE) end
+	end; oFTrc.mFlt.Size = (iE - 1)
+	return oFTrc
+end
+
+--[[
+ * Moves only the entities from source to destination
+ * oFTrc > Reference to tracer object
+ * tData > Source data table to read from ( entity array )
+ * bTab  > When enabled process as table instead of array
+ * bID   > When enabled the source table contains entity ID
+]]
+local function putFilterEar(oFTrc, tData, bTab, bID)
+	if(not oFTrc) then return nil end
+	local tE = oFTrc.mFlt.Ear
+	if(bTab) then local iD = 1
+		while(tData[iD]) do

```

Comment:
Why a while loop here?

Reply:
Hi, because I don't know if the `entity` or `entity ID` are valid ot not before adding them to the table. I do not know where the array ends in advance, so just keep incrementing and  validating until the `tE = oFTrc.mFlt.Ear` is populated with all validated values.

Reply:
What do you mean you don't know where the array ends? Shouldn't `#tData` be the end?

Reply:
It will, but it will count all the elements and I want to avoid that.

Reply:
It is counting all the elements anyway using iD isn't it?

Reply:
Yeah. But loops in one pass, counts them and process them. ;)

Reply:
It's slower to do it this way in lua. Use ipairs

Reply:
It is only done when the filter type is switched during initialization aniway ... Calling of ipairs is [deprecated](https://www.google.com/search?q=ipairs+is+depricated&oq=ipairs+is+depricated).

Reply:
No it's not

Reply:
This is gonna be executed only when the `ftrace` object filter is changed to `entity array` so this is designed to be called in `if(first() || dupefinished())`